language/golang/mistakes_test.go

```diff
@@ -0,0 +1,97 @@
+package golang
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/symflower/eval-dev-quality/log"
+	"github.com/zimmski/osutil"
+)
+
+func TestMistakes(t *testing.T) {
+	type testCase struct {
+		Name string
+
+		RepositoryPath string
+
+		ExpectedMistakes []string
+	}
+
+	validate := func(t *testing.T, tc *testCase) {
+		t.Run(tc.Name, func(t *testing.T) {
+			temporaryPath := t.TempDir()
+			repositoryPath := filepath.Join(temporaryPath, filepath.Base(tc.RepositoryPath))
+			require.NoError(t, osutil.CopyTree(tc.RepositoryPath, repositoryPath))
+
+			buffer, logger := log.Buffer()
+			defer func() {
+				if t.Failed() {
+					t.Log(buffer.String())
+				}
+			}()
+
+			actualMistakes, actualErr := Mistakes(logger, repositoryPath)
+			require.NoError(t, actualErr)
+
+			assert.Equal(t, tc.ExpectedMistakes, actualMistakes)
+		})
+	}
+
+	validate(t, &testCase{
+		Name: "Function without opening bracket",
+
+		RepositoryPath: filepath.Join("..", "..", "testdata", "golang", "mistakes"),
+
+		ExpectedMistakes: []string{
+			"." + string(os.PathSeparator) + "functionWithoutOpeningBracket.go" + ":3:1: syntax error: non-declaration statement outside function body",
```

Comment:
`filepath.Join(".","functionWithoutOpeningBracket.go")`

Reply:
That does not work, `filepath.Join(".","functionWithoutOpeningBracket.go") == functionWithoutOpeningBracket.go` and not `./functionWithoutOpeningBracket.go`

Reply:
oh right because the result is cleaned.